sdk/core/Azure.Core/src/Shared/OperationInternalOfT.cs

```diff
@@ -274,16 +279,74 @@ protected override async ValueTask<Response> UpdateStatusAsync(bool async, Cance
             }
         }
 
-        private static Response GetResponseFromState(OperationState<T> state)
+        private static Response GetResponseFromState(OperationState<T> state, RequestMethod? requestmethod = null)
         {
             if (state.HasSucceeded)
             {
                 return state.RawResponse;
             }
 
+            // if this is a fake delete lro with 404, just return empty response with 204
+            if (RequestMethod.Delete == requestmethod && state.RawResponse.Status == 404)
+            {
+                return new EmptyResponse(HttpStatusCode.NoContent);
+            }
+
             throw state.OperationFailedException!;
         }
 
+        private class EmptyResponse : Response
```

Comment:
Ideally, we would avoid adding new internal subtypes of `Response`.  Is there one in the Azure.Core code base you can reuse?  If not, it would be good to add a comment to the type to explain why it exists and why it is unique, when you would expect to return it to an end-user, and what they would expect from it, so that maintainers of this code no what they can and can't change going forward.

Reply:
This is only used for fake delete LRO, we just want to return an empty response with 204 to the user for this case.

Added the above comment to the class documentation. I can't find any existing internal implemented `Response` type for this usage.